### PR TITLE
add variableName option

### PR DESCRIPTION
**Pull request by ywmail**
add option `variableName` for another choice.

### CODE REVIEW

**Comment by agoldis** [comment]
@ywmail great work on PR! I will review and merge during the upcoming weekend.

File: src/plugin.js [hunk 1]
```diff
@@ -70,6 +75,9 @@ class WebpackRequireFrom {
         "Object.defineProperty(" + mainTemplate.requireFn + ', "p", {',
         "  get: function () {",
         getterBody,
+        " },",
+        "  set: function (value) {",
+        "",
```
Comment (agoldis) [inline src/plugin.js]: Curious why did you add a setter?

File: src/codeBuilders.js [hunk 2]
```diff
@@ -41,3 +41,20 @@ exports.buildMethodCode = function(methodName, defaultPublicPath, shouldSupressE
...
 };
+
+
+exports.buildVariableCode = function (variableName, defaultPublicPath, shouldSupressErrors = false) {
+  return [
+    "try {",
+    `  if (typeof ${variableName} !== "string") {`,
+    `    throw new Error("${PLUGIN_NAME}: '${variableName}' is not a string or not available at runtime. See https://github.com/agoldis/webpack-require-from#troubleshooting");`,
+    "  }",
+    `  return (function(){if($PIX) return ${variableName};})();`,
```
Comment (agoldis) [inline src/codeBuilders.js]: `$PIX`?

**Comment by ywmail** [comment]
Thanks for the review and your great work, it's really a great plugin. 💯 
I'll update the PR later, it's my first PR on github. :) 🔢

File: src/plugin.js (hunk 1 above)
Comment (ywmail) [reply src/plugin.js]: Because webpack allow users to change the `__webpack_public_path__` in their code for a dynamic outputPath, if their is no setter in property `p`, the application will fail with a error message:
`Cannot set property p of function...`
See: 
https://webpack.js.org/guides/public-path/#on-the-fly
so i think it's better to add a setter to warn the user in the setter function like:
```
        "  set: function (value) {",
        `console.warn("You should not change __webpack_public_path__ directly, use ${PLUGIN_NAME} to config public path.")`,
        " }",
```

File: src/codeBuilders.js (hunk 2 above)
Comment (ywmail) [reply src/codeBuilders.js]: `$PIX` is the global object in my project, I committed this by mistake, will remove it.

File: src/plugin.js (hunk 1 above)
Comment (ywmail) [reply src/plugin.js]: Do you think it's necessary?

Comment (ywmail) [reply src/plugin.js]: @agoldis 
Yes, I also think it's better to handle this scenario in separate issue gently, a warning in setter function is just temporary in my case. I will remove this part from the PR. :)

**Comment by ywmail** [comment]
@agoldis 
Update the PR: add some tests to confirm that `variableName` works as expected and doesn't break the original function. :)
A great thanks for reviewing the code. 💯
